runtime/evaluate.cc

```diff
@@ -496,9 +501,9 @@ class evaluator {
       // Running the worker here guarantees forward progress on the loop even if no threads in the thread pool are
       // available.
       pool->run(worker, loop.get());
-      // While the loop still isn't done, work on other tasks.
-      pool->wait_for([&]() { return loop->done(); });
 
+      // While the loop still isn't done, work on other tasks.
```

Comment:
How does this work now? It doesn't work on other tasks any more does it?

I think that's a pretty important capability, for both performance and might also be a deadlock risk (this implementation "burns" a thread, so we might run out of threads to do work with).

Reply:
no, it doesn't.
I've realized this might change might not be a good idea.